### ISSUE TITLE
Accessibility: Hidden windows present themselves as active to Linux screenreaders

### ISSUE THREAD

**Issue by RastislavKish** [electron/electron#48268]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

38

### What operating system(s) are you using?

Other Linux

### Operating System Version

openSUSE Tumbleweed 20250904

### What arch are you using?

x64

### Does the issue also appear in Chromium / Google Chrome?

I don't know how to test

### Expected Behavior

When calling the .hide() method on a BrowserWindow instance, the window disappears from the screen, and should also fire a window deactivate accessibility event to screenreaders through at-spi. When queried for its state by a running screenreader, the window should report itself as inactive until it's shown again. These mechanisms allow the screenreaders to determine which window is currently focused, and provide the user with appropriate feedback and various navigation tools.

### Actual Behavior

After calling the .hide() method, the window disappears, but the deactivation a11y event is not fired and when queried by the assistive tech, the window presents itself as active despite it's hidden.

This is a quite severe problem for Linux screenreaders. The user may be running various popular Electron apps like Signal, Element or Discord in the background, but these all report themselves as active to the screenreader, which afterwards often reads a completely different window to the user from the one they actually want to use, because it's unable to determine which of them is really focused.

### Testcase Gist URL

https://gist.github.com/RastislavKish/c9ec92351be7c9e77d965e601465f4c9

### Additional Information

Reproduction steps

Detecting the status of windows with a screenreader directly is complicated, because Orca, the Linux screenreader, applies some mitigations against this problem, and so the results can be somewhat unpredictable. But there is an official Orca script which prints plain at-spi window activations and deactivations to the terminal, as well as a list of windows claiming to be active. This python script only requires at-spi2-core-dev libraries and the gobject introspection libraries, which are both included in the repositories of all common Linux distributions.

Steps to reproduce:

1. Clone the official Orca repository and cd into the tools directory:

```sh
git clone https://gitlab.gnome.org/GNOME/orca.git
cd orca/tools
```

2. Turn on Orca, so we can be sure the Electron a11y stack starts properly.
3. Start Electron fiddle, and load the attached gist. This gist is the standard hello world app, which after 5 seconds upon launch calls mainWindow.hide()
4. Return to the terminal, and launch:

```sh
python3 dump-window-events.py
```

5. Switch the window, either by mouse or by using Alt+Tab, back to the Electron fiddle and hit F5 to run the gist.
6. Wait until the Hello world app launches, stays for 5 seconds and disappears.
7. Switch the window, either by mouse or pressing Alt+Tab, back to the terminal with the running script. The script should notice the terminal has been focused and exit automatically. Check the output.

In the output, you should see the Hello world app fires the activation event when it first appears on the screen, but then does not fire the deactivation event when it hides, and when switching windows, it keeps reporting itself as being active, even though only one window should always be active at once.

I'm attaching the output of the command captured on my machine running GNOME 48.4 Wayland. Ignore everything from mutter-x11-frames, what matters for Orca are the events from the real apps.

Note: If you don't see *any* accessibility events from the electron apps, i.e. electron-fiddle and the Hello world app, it may be necessary to set the ACCESSIBILITY_ENABLED environment variable to 1, and if it doesn't help, also use --force-renderer-accessibility flag with the Hello world app. Neither of these should be necessary by default when Orca is turned on, but there are certain setups where the AT auto-detection mechanism fails for various reasons.

[dump-window-events-output.log](https://github.com/user-attachments/files/22191937/dump-window-events-output.log)

**Comment by alex19EP** [comment]
can reproduce on electron [39.2.6](https://releases.electronjs.org/release/v39.2.6)

**Comment by RastislavKish** [comment]
Still present in 41.0.2

**Comment by RastislavKish** [comment]
With Gnome 50, at-spi2-core-2.60.4 and Orca 51.alpha, I'm no longer able to reproduce this since at least 40.10.3!
